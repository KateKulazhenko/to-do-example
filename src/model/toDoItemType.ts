export interface ToDoItemEntity {
  id: number;
  isDone: boolean;
  text: string;
}
