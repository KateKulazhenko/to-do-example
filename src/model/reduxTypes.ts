import { ToDoItemEntity } from './toDoItemType';
import { Action } from 'redux';

export type State = {
  toDoList: ToDoItemEntity[],
  index: number,
  error: {} | null
};

export interface AddItemAction extends Action {
  type: 'ADD_ITEM';
  payload: {
    text: string;
  };
};

export interface ChangeStatusAction extends Action {
  type: 'CHANGE_STATUS';
  payload: {
    id: number;
  };
};

export interface RemoveItemAction extends Action {
  type: 'REMOVE_ITEM';
  payload: {
    id: number;
  };
};

export type ToDoActions = AddItemAction & ChangeStatusAction & RemoveItemAction;