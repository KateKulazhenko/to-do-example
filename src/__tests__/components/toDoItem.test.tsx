import * as React from 'react';
import {shallow} from 'enzyme';

import {ToDoItemEntity} from 'model/toDoItemType';
import ToDoItem, { Props } from 'components/toDoList/toDoItem/toDoItem';

describe ('components/toDoList/toDoItem tests', () => {
  it('should render as expected ', () => {
      // Arrange
      const newItem: ToDoItemEntity = {
        id: 0,
        text: 'test text',
        isDone: false
      };

      const testingProps: Props = {
        toDoItem: newItem,
        onStatusChanged: () => {return; },
        onRemove: () => {return; }
      };

      // Act
      const component = shallow(
          <ToDoItem {...testingProps} />
       );
      // Assert
      expect (component).toMatchSnapshot();
  });
});
