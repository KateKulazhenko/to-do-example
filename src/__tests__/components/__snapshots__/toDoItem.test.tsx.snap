// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/toDoList/toDoItem tests should render as expected  1`] = `
<li
  className="to-do-list__to-do-item to-do-item"
>
  <input
    checked={false}
    className="to-do-item__checkbox"
    id="0"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="to-do-item__label"
    htmlFor="0"
  />
  <p
    className="to-do-item__text"
  >
    test text
  </p>
  <button
    aria-label="Remove item with text: test text"
    className="to-do-item__remove-button"
    onClick={[Function]}
  >
    x
  </button>
</li>
`;
